src/main/java/com/spikes2212/control/RioPIDLoop.java

```diff
@@ -0,0 +1,180 @@
+        this(kp, ki, kd, tolerance, setpoint, PIDSource, speedConsumer, waitTime, Frequency.DEFAULT);
+    }
+
+    public RioPIDLoop(Supplier<Double> kp, Supplier<Double> ki, Supplier<Double> kd, Supplier<Double> tolerance,
+                      Supplier<Double> setpoint, Supplier<Double> PIDSource, Consumer<Double> speedConsumer, Frequency frequency) {
+        this(kp, ki, kd, tolerance, setpoint, PIDSource, speedConsumer, () -> 0.0, frequency);
+    }
+
+    public RioPIDLoop(Supplier<Double> kp, Supplier<Double> ki, Supplier<Double> kd, Supplier<Double> tolerance,
+                      Supplier<Double> setpoint, Supplier<Double> PIDSource, Consumer<Double> speedConsumer) {
+        this(kp, ki, kd, tolerance, setpoint, PIDSource, speedConsumer, () -> 0.0, Frequency.DEFAULT);
+    }
+
+    public RioPIDLoop(double kp, double ki, double kd, double tolerance,
+                      double setpoint, Supplier<Double> PIDSource, Consumer<Double> speedConsumer, double waitTime, Frequency frequency) {
+        this(() -> kp, () -> ki, () -> kd, () -> tolerance, () -> setpoint, PIDSource, speedConsumer, () -> waitTime, frequency);
+    }
+
+    public RioPIDLoop(double kp, double ki, double kd, double tolerance,
+                      double setpoint, Supplier<Double> PIDSource, Consumer<Double> speedConsumer, double waitTime) {
+        this(() -> kp, () -> ki, () -> kd, () -> tolerance, () -> setpoint, PIDSource, speedConsumer, () -> waitTime, Frequency.DEFAULT);
+    }
+
+    public RioPIDLoop(double kp, double ki, double kd, double tolerance,
+                      double setpoint, Supplier<Double> PIDSource, Consumer<Double> speedConsumer, Frequency frequency) {
+        this(() -> kp, () -> ki, () -> kd, () -> tolerance, () -> setpoint, PIDSource, speedConsumer, () -> 0.0, frequency);
+    }
+
+    public RioPIDLoop(double kp, double ki, double kd, double tolerance,
+                      double setpoint, Supplier<Double> PIDSource, Consumer<Double> speedConsumer) {
+        this(() -> kp, () -> ki, () -> kd, () -> tolerance, () -> setpoint, PIDSource, speedConsumer, () -> 0.0, Frequency.DEFAULT);
+    }
+
+    @Override
+    public void enable() {
+        controller = new PIDController(kp.get(), ki.get(), kd.get(), frequency.period);
+        notifier = new Notifier(this::periodic);
+        notifier.startPeriodic(frequency.period);
+    }
+
+    private void periodic() {
+        PIDCalculations = () -> controller.calculate(PIDSource.get(), setpoint.get());
+    }
+
+    @Override
+    public void disable() {
+        notifier.stop();
+        notifier.close();
+    }
+
+
+    public void enableContinuousInput(double minimumValue, double maximumValue) {
+        this.controller.enableContinuousInput(minimumValue, maximumValue);
+    }
+
+    @Override
+    public void update() {
+        speedConsumer.accept(PIDCalculations.get());
```

Comment:
replace current `update` method with a method that uses `controller.setSetpoint` and is synchronized on `controller`